Updated extended manual deployment instructions for Ubuntu 20.04

sipsorcery opened the pull request:
Updated extended manual deployment instructions for Ubuntu 20.04 and latest versions of application components (btc, lnd and rtl).

Adjusted steps where necessary for changes to deployment instructions for each application.

Changed the Bitcoin Core instructions to build from source rather than download. WIth this change the instructions now document a full build from source solution for BTCPay Server.

dennisreimann commented on docs/ManualDeploymentExtended.md:
```diff
@@ -416,14 +451,13 @@ Create a bitcoin.conf file to suit your needs. An example file that is suitable
 rpcbind=127.0.0.1                     # loopback is default for 0.18.0 but no harm making sure.
 whitelist=127.0.0.1                   # for nbxplorer.
 rpcallowip=127.0.0.1/32               # loopback is default but again no harm.
-disablewallet=1                       # btcpay does not host a bitcoin wallet.
```
Why remove that?

NicolasDorier replied:
Because we do not disable the wallet anymore, even on the docker deploy

dennisreimann commented on docs/ManualDeploymentExtended.md:
````diff
@@ -122,17 +124,12 @@ Tor can be used by the following components to provide enhanced privacy and/or h
 ##### 🚚 Install
 
 ```bash
-~$ sudo apt install apt-transport-https
-~$ sudo vi /etc/apt/sources.list # (and append two lines below)
-deb https://deb.torproject.org/torproject.org bionic main
-deb-src https://deb.torproject.org/torproject.org bionic main
-~$ curl https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc | gpg --import
-~$ gpg --export A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89 | sudo apt-key add -
-~$ sudo apt update
-~$ sudo apt install tor deb.torproject.org-keyring
````
Isn't this necessary anymore? Maybe it'd make sense to optionally include this for previous Ubuntu versions?

NicolasDorier replied:
I think that's good we just put things for the latest version if this is simpler.

sipsorcery replied:
The recommendation on the Tor website is to use "apt install" (I can't recall the reason I didn't use that last time but most likely it was the Ubuntu 18.04 package not supporting Torv3).

dennisreimann commented on docs/ManualDeploymentExtended.md:
````diff
@@ -353,17 +356,36 @@ Congratulations, all renewals succeeded. The following certs have been renewed:
 ```
 
 ```bash
 ~$ sudo sudo netstat -tlnp | grep nginx
````
Double `sudo` for god mode? 😏

dennisreimann commented on docs/ManualDeploymentExtended.md:
```diff
@@ -960,6 +1063,10 @@ Confirm wallet password:
 21. resemble  22. angry   23. work     24. until
 ---------------END LND CIPHER SEED-----------------
+
+!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO RESTORE THE WALLET!!!
```
Maybe it makes sense to highlight this even more using a ```:::warn``` block.

sipsorcery replied:
That's actually just a copy of the console output from running the "lncli create" command. I guess another warn block in the doc could be added but probably redundant given they'll get the same double warning when they run the command?

sipsorcery commented:
> One thing I'd like to discuss: Do we want to propose building everything from source?

The only application component that previously wasn't being built from source was `Bitcoin Core`. I don't think it matters a lot whether `Bitcoin Core` is built or downloaded from a trusted source. I switched to using the build from source for completeness, since that's what all the other components were doing. 

The type of user likely to follow the steps is likely to be fairly technical and will be able to switch approaches as they desire. I've added the steps as a time saver for hunting through other documents.

NicolasDorier commented:
Yeah I think this doc is very important for linux people to understand how things interact with each other. As @sipsorcery said, they are savvy enough to tweak things as they want.
